src/main/java/org/opensearch/ad/transport/SearchTopAnomalyResultTransportAction.java

```diff
@@ -308,6 +308,12 @@ protected void doExecute(Task task, SearchTopAnomalyResultRequest request, Actio
             // Generating the search request which will contain the generated query
             SearchRequest searchRequest = generateSearchRequest(request);
 
+            // Adding search over any custom result indices
+            String customResultIndex = Strings.trimToNull(getAdResponse.getDetector().getResultIndex());
+            if (customResultIndex != null) {
+                searchRequest.indices(defaultIndex, customResultIndex);
```

Comment:
TODO: if an index pattern is set here, this may fail. May need to search and iterate over existing indices that match the detector, and add all of them into a concrete list of custom result indices to search over. See SearchAnomalyResultTransportAction in #276.

Reply:
^ this can probably be ignored. Since we don't need to add a result index param in the API (which could contain a wildcard / be an index pattern), then we can strictly retrieve the single concrete result index from the detector.

Reply:
What is the reason for still needing to search all AD Result indices if we know the detector has a custom result index?

Reply:
it's following the same strategy as custom result index implementation, which searches over both, as a performance tradeoff so that we don't miss any results, in queries where we want to fetch results from both places (e.g., detector list fetches results from detectors that have results in both system index & custom result indices). 

If in the future we let user change the destination result index, we would want to support both/multiple

Reply:
^ also, its a very minor performance hit here, since the result scope is narrowed to only one detector + results w/ anomaly grade > 0

Reply:
Got it thank you